src/scheduler.cc

```diff
@@ -456,7 +460,28 @@ std::pair<WorkerId, ObjStoreId> SchedulerService::register_worker(const std::str
     } else {
       (*workers)[workerid].current_task = NO_OPERATION;
     }
+
+    if (!is_driver) {
+      // Import remote functions on the worker.
+      for (int i = 0; i < exported_functions.size(); ++i) {
+        ClientContext import_context;
+        ImportFunctionRequest import_request;
+        import_request.mutable_function()->CopyFrom(*exported_functions[i].get());
+        ImportFunctionReply import_reply;
+        (*workers)[i].worker_stub->ImportFunction(&import_context, import_request, &import_reply);
+      }
+
+      // Import reusable variables on the worker.
+      for (int i = 0; i < exported_reusable_variables.size(); ++i) {
+        ClientContext import_context;
+        ImportReusableVariableRequest import_request;
+        import_request.mutable_reusable_variable()->CopyFrom(*exported_functions[i].get());
+        AckReply import_reply;
+        (*workers)[i].worker_stub->ImportReusableVariable(&import_context, import_request, &import_reply);
+      }
+    }
```

Comment:
Yes, they should be separate methods. I'm not sure I understand the last part, can you elaborate?